protos/core/core.proto

```diff
@@ -36,3 +40,18 @@ message SetMavlinkTimeoutResponse {}
 message ConnectionState {
     bool is_connected = 2; // Whether the vehicle got connected or disconnected
 }
+
+// Defines a request to list components of all systems.
+message ListComponentsRequest {}
+
+// Defines a single system's components.
+message SystemComponents {
+  uint32 system_id = 1; // System ID
+  repeated uint32 component_ids = 2; // List of component IDs for this system
+}
+
+// Defines the response for a request to list components of all systems.
+// It includes a list of systems, each with their own list of component IDs.
+message ListComponentsResponse {
+  repeated SystemComponents systems = 1; // List of systems with their components
```

Comment:
Hey Julian,

In my Discord comment, I mentioned that I had one system from which I could list all components since my feature only needs one system. However, in my merge request, I extended it to include listing all systems because I'm preparing to share a draft with you on how to select systems over gRPC. But of course, if it's more convenient for you to wait for the bigger MR draft, then feel free to ignore this one.

Reply:
Got it, thanks! I will have to think about this a bit more in the coming weeks, and talk to @JonasVautherin to figure out how we want to roll this out to the language wrappers.

Reply:
Maybe it would be a good idea if @NourElMenshawy described the idea somewhere (maybe in an issue). Whatever is done has to be compatible with the language wrappers (that's the whole point of the gRPC layer, after all :innocent:).

Reply:
Hello @JonasVautherin 
The purpose of this MR is to retrieve all components from the MAVSDK server over gRPC. I developed a Dart application that reads parameters from the drone via an LTE server. In this setup, the LTE server communicates with MAVLink, and my custom MAVSDK server includes the gRPC method for getting all components. My Dart application reads and writes parameters to the drone. Being able to select components allows my application to set and read the parameters of those components.

Reply:
Sure, but listing MAVLink systems is not the same as being able to talk to different systems. Say in Python you do `system.action.arm()`, how do you deal with two systems?

My personal opinion tends to be that you can demultiplex the MAVLink stream and run multiple instances of MAVSDK. So if some messages come from sysid 12 and some from sysid 13, you can have your demultiplexer send sysid 12 to some local port (and bind the first MAVSDK instance to it) and have it send sysid 13 to some other local port (and bind the second MAVSDK instance to it).

Reply:
Hello @jona
I do read my comments and I think I didn't clearly convey my point, so I'll try again:

1. I don't care about the system ID because I only have one drone. I made the system ID a list to potentially support multiple systems in the future.
2. I need to read the parameters connected to the drone, including those from external components, but I don't know their IDs. Therefore, I need MAVSDK to provide me with those component IDs so I can select them and read their parameters.